### DISCUSSION TITLE
OS X and Windows VM

### DISCUSSION THREAD

**Discussion by ModMike** [JPersson77/LGTVCompanion#228] [Q&A] [answered]
I just got the 42" C3 to use as a monitor. The thing is, I am on OS X but am running a windows VM. I can also boot into windows directly if needed.

From what I see, you only need to run the companion app to set up the TV, then everything happens locally on the tv. Is this correct or does the Companion App need to run all the time?

Thanks for the great work!

**Comment by ModMike** [comment]
Just read a little more carefully and see that it starts a service. So app is running all the time, which makes sense, correct? Is there anyway to compile this for OS X?

**Comment by JPersson77** [comment, marked as answer]
Hi, yes the service is running in the background and intercept power messages from windows. There is also a user mode application which monitors other aspects and runs continually.

There is some interest for porting but the app is windows only at the moment. Some key components are already portable but some other components rely on windows specifics.

**Comment by ModMike** [comment]
Is it correct to assume that the service needs to run on the device connected to the TV to read the device states? If not, can I virtualize it on my network appliance?

**Comment by JPersson77** [comment]
The service needs to run on the PC which controls the TV, which is typically the PC connected to the TV. Both the service and daemon are integrated and required for the app's normal functining. Apart from the service and the mentioned daemon, there is also a console application which does not rely on any other executable.

The service can not really be run on another  network appliance since it does not only communicate with the TV, it also intercept windows messages etc.

Thinking freely it would probably be possible to compile a portable exe of the console app (lgtvcli.exe) and use that in conjunction with chron scripts etc. But there are other libaries/apps in python which might be a better fit then

**Comment by ModMike** [comment]
Thanks for the clarification!

**Comment by JPersson77** [comment]
For anyone else reading this thread and thinking about contributing. I have split the original codebase into more code modules, from v 4 and onwards. 

Thiw means that for example the core logic and the webos client is now portable, as is. Other modules like preferences management and intra-process comms should be portable with some minor modifications.

The windows specifics in the service and the daemon are what needs more work obviously. Intercepting power messages, user idle status event log messages etc. And replicating the UI/configuration which write config.json

**Comment by ModMike** [comment]
I be been away recovering from a double bypass. Has there been any takers for is x?

**Comment by JPersson77** [comment]
Good to hear you are recovering! None that I'm aware of is currently working on a pprt

**Comment by ModMike** [comment]
I’m wondering if moving it to home assistant would help things, basically like David Zhang. Seems to eliminate the code running on the actual device dependency. 

https://youtu.be/76wAEJOdBq8?si=8DPXl73_OzIZF0wj

frankly and decent me sensor would work and basically make it plug and play for home assistant users. The Aqara sensor would be a good choice.

**Comment by JPersson77** [comment]
Yeah that looks nice and functional. Been thinking about integration points with mqtt to allow for motion presence detection. I like the idea of being able to extend the user idle mode detection (without hass.agent)

**Comment by ModMike** [comment]
Looking at EMQX. The Add-on for Home Assistant is a full-featured MQTT Broker. It’s 

Would be the cleanest way for me to control monitor rather than check if I’m in my chair, although that is probably best mode. Would have added advantage of automation to fully shut off monitor when it’s powered down for some time, as well as other functions I need for lights and webcams.

i will work on I tonight if I feel better, still trying to understand the best path and tools to use.
